File: app/src/main/java/com/example/Sleepy/adapters/OnBoardingAdapter.java
package com.example.Sleepy.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.airbnb.lottie.LottieAnimationView;
import com.example.Sleepy.R;

import java.util.List;

public class OnBoardingAdapter extends RecyclerView.Adapter<OnBoardingAdapter.ViewHolder> {
    private final List<OnBoarding> boardings;
    View view;

    public OnBoardingAdapter(List<OnBoarding> states) {
        this.boardings = states;
    }

    @Override
    public OnBoardingAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_container_onboarding, parent, false);
        return new OnBoardingAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(OnBoardingAdapter.ViewHolder holder, int position) {
        OnBoarding state = boardings.get(position);
        holder.titleView.setText(state.getTitle());
        holder.descText.setText(state.getDescription());
        holder.lottie.setImageResource(state.getImage());
    }

    @Override
    public int getItemCount() {
        return boardings.size();
    }

    public void delete(int position){
        boardings.remove(position);
        notifyItemRemoved(position);
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        final TextView titleView, descText;
        final LottieAnimationView lottie;
        ViewHolder(View view){
            super(view);
            titleView = view.findViewById(R.id.tvTitleBoard);
            descText = view.findViewById(R.id.tvDescriptionBoard);
            lottie = view.findViewById(R.id.lottieSrcBoard);
        }
    }
}

File: app/src/main/java/com/example/sleepy/presentation/onBoarding/viewpager/OnBoardingAdapter.kt
package com.example.sleepy.presentation.onBoarding.viewpager

import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.sleepy.R
import android.widget.TextView
import com.airbnb.lottie.LottieAnimationView

class OnBoardingAdapter(private val boardings: ArrayList<OnBoarding>) :
    RecyclerView.Adapter<OnBoardingAdapter.ViewHolder>() {
    lateinit var view: View

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        view = LayoutInflater.from(parent.context).inflate(R.layout.item_container_on_boarding, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val state = boardings[position]
        holder.titleView.text = state.title
        holder.descText.text = state.description
        holder.lottie.setImageResource(state.image)
    }

    override fun getItemCount(): Int {
        return boardings.size
    }

    fun delete(position: Int) {
        boardings.removeAt(position)
        notifyItemRemoved(position)
    }

    class ViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(
        view
    ) {
        val titleView: TextView = view.findViewById(R.id.tvTitleBoard)
        val descText: TextView = view.findViewById(R.id.tvDescriptionBoard)
        val lottie: LottieAnimationView = view.findViewById(R.id.lottieSrcBoard)
    }
}